File: PagingRoomJava/app/src/main/java/com/websarva/wings/android/pagingroomjava/data/local/PhoneDAO.java
package com.websarva.wings.android.pagingroomjava.data.local;

import androidx.paging.PagingSource;
import androidx.room.Dao;
import androidx.room.Query;

@Dao
public interface PhoneDAO {
	@Query("SELECT * FROM phones ORDER BY id")
	PagingSource<Integer, Phone> findAll();
}

File: PagingRoomKotlin/app/src/main/java/com/websarva/wings/android/pagingroomkotlin/data/local/PhoneDAO.kt
package com.websarva.wings.android.pagingroomkotlin.data.local

import androidx.paging.PagingSource
import androidx.room.Dao
import androidx.room.Query

@Dao
interface PhoneDAO {
	@Query("SELECT * FROM phones ORDER BY id")
	fun findAll(): PagingSource<Int, Phone>
}
